Tools: Add additional options for dumping in GS runner.

TJnotJT opened the pull request:
### Description of Changes
Add additional options to GS dump runner to allow: dumping render target, textures, depth, alphas per draw; range of frames or draws to dump. Some miscellanous fixes to formatting of file names and dumping of GS context.

### Rationale behind Changes
To make it easier to debug large numbers of GS dumps. Example workflow: run a large number of GS dumps just outputting the frames, check which dumps/frames are different, rerun the dump runner on the differing dumps/frames while dumping all draws RTs to see which draw the change ocurred.

### Suggested Testing Steps
Run the GS dump runner with new options -dump [rt|tex|z|f|a|i], -dumprange, -dumprangef. Example:
`pcsx2-gsdumprunner-x64-dbg.exe <dump file> -renderer sw -dumpdir <output dir> -dump rttexfzai -dumprange 100,100,3`
to dump RTs, textures, frames, alphas,and info (context, vertex, vsync regs) from in draws 100 to 199 that are multiples of 3.

F0bes reviewed (approved):
Code looks fine. Does @lightningterror want to do a test dump, just to see if the basic functionality is still working properly? :)

TJnotJT commented on pcsx2-gsrunner/Main.cpp:
```diff
@@ -443,6 +443,14 @@ static void PrintCommandLineHelp(const char* progname)
 	std::fprintf(stderr, "  -help: Displays this information and exits.\n");
 	std::fprintf(stderr, "  -version: Displays version information and exits.\n");
 	std::fprintf(stderr, "  -dumpdir <dir>: Frame dump directory (will be dumped as filename_frameN.png).\n");
+	std::fprintf(stderr, "  -dump [rt|tex|z|f|a|i]: Enabling dunmping of render target, texture, z buffer, frame, "
```
Thanks for the review Fobes. Fixed the typo.

Mrlinkwii commented:
just to note their are conflicts

TJnotJT commented:
Thanks for the head up @Mrlinkwii. I fixed the conflict, which was a blank line. I also made a small formatting fix, which apparently reverts something in a PR since I made this one:

Old:
`s = GetDrawDumpPath("%05d_f%lld_rt1_%05x_(%05x)_%s.bmp", s_n, frame, m_cached_ctx.FRAME.Block(), rt->m_TEX0.TBP0, psm_str(m_cached_ctx.FRAME.PSM));`

New:
`s = GetDrawDumpPath("%05d_f%05lld_rt1_%05x_%s.bmp", s_n, frame, m_cached_ctx.FRAME.Block(), psm_str(m_cached_ctx.FRAME.PSM));`

The parenthesis around block number are inconsistent with the way SW renderer names dump files and others names even in HW renderer, so I changed it back. Hope that's ok.

lightningterror commented:
This breaks draw dumping on opengl, no context or vertex info is dumped.

TJnotJT commented:
> This breaks draw dumping on opengl, no context or vertex info is dumped.

Should be fixed by adding missing UI elements in https://github.com/PCSX2/pcsx2/pull/12302
